lib/litmus_paper/cache.rb

```diff
@@ -11,12 +11,18 @@ def initialize(location, namespace, ttl)
 
     def set(key, value)
       return unless @ttl > 0
-      File.open(File.join(@path, key), "a") do |f|
-        f.flock(File::LOCK_EX)
-        f.rewind
-        f.write("#{Time.now.to_f + @ttl} #{YAML::dump(value)}")
-        f.flush
-        f.truncate(f.pos)
+      filepath = File.join(@path, key)
+      begin
+        File.open(filepath, "r+") do |f|
+          f.flock(File::LOCK_EX)
+          f.rewind
+          f.write("#{Time.now.to_f + @ttl} #{YAML::dump(value)}")
+          f.flush
+          f.truncate(f.pos)
+        end
+      rescue Errno::ENOENT => e
+        File.open(filepath, "a") {}
+        set(key, value)
```

Comment:
This rescue seems awkward from a readability standpoint. Also, creating the file with `File.open(filepath, "a") {}` means we have an empty file that other processess could try to read and parse. Perhaps a better option would be:
```
if cache file exists; then
  lock and write
else
  create temp file on the same file system
  write value
  rename temp file to filepath
fi
```

Reply:
Yeah, I'm not super happy with the current implementation. My initial thought process was that the file would only not exist in an exceptional case, and we wouldn't want to check if it exists on every `set` call, but only if we run into an error. 

The atomicity of writing to a temp file and then renaming seems better then simply creating and writing.

Any thoughts on keeping the exception handling, but updating the creation code?

Reply:
You are already checking file existence on every get call. Also my understanding is that checking for file existence on a file that is in file cache should be very fast.

I would propose checking for file existence in set, and explicitly creating the file there if it does not exist. On startup, there is a possibility that multiple unicorn processes will create the file, but that should be a small onetime cost.